### DISCUSSION TITLE
taxon_collapser.py

### DISCUSSION THREAD

**Discussion by coreyholt** [TheBrownLab/PhyloFisher#47] [Q&A] [answered]
Hi there, 

Where in the pipeline would I use taxon_collapser.py? And is this different from the chimera options in taxa and gene selection?

Cheers!
Corey

**Comment by atice** [comment, marked as answer]
Hi Corey,

Typically, taxon_collapser.py is used after apply_to_db.py. The way it differs from the chimera option you mentioned is that taxon_collapser.py will create a permanent entry in your database with its own unique ID that represents the most complete proteome for all taxa collapsed to make it. While the chimera option will not create a permanent database entry. The chimera produced through this option will exist only in your matrix and single gene alignments for the  phylogenomic dataset you created when you invoked the chimera flag.

Often times in our lab students will generate several single cell transcriptomes from the same population of an "uncultivable" microbe. After we are sure the cells all represent the same species we use taxon_collapser.py to create a permanent most-complete proteome of that organism for use in subsequent studies.

Does this clarify? Let me know if you have any further questions and thank you for your interest in PhyloFisher.

Alex

Alexander K. Tice, Ph.D.
Department of Biological Sciences
Mississippi State University
027 Harned Hall
Lab Telephone: 662-325-4008

________________________________
From: coreyholt ***@***.***>
Sent: Wednesday, October 5, 2022 1:25 PM
To: TheBrownLab/PhyloFisher ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [TheBrownLab/PhyloFisher] taxon_collapser.py (Discussion #47)


Hi there,

Where in the pipeline would I use taxon_collapser.py? And is this different from the chimera options in taxa and gene selection?

Cheers!
Corey

**Reply by coreyholt** [reply]
Got it. Thank you, Alex!

**Comment by atice** [comment]
Hey Corey,

You should see the file select_taxa.tsv that was created after you ran the script. If it is there then you should be able to continue down the pipeline with no problem. The error message you received was because the script was unable to produce a plot due to some weird panda's issue that shows up when the number of taxa you selected is not divisible by 5.  Robert went ahead and added a try-except state where in the future the script will print a friendly message that it could not produce the plot if it can't until this issue is resolved. Hope this helps. Let us know.

Alex and Robert

Alexander K. Tice, Ph.D.
Department of Biological Sciences
Mississippi State University
027 Harned Hall
Lab Telephone: 662-325-4008

________________________________
